store/store.go

```diff
@@ -1534,7 +1534,14 @@ func downloadImpl(ctx context.Context, name, sha3_384, downloadURL string, user
 			}
 			break
 		}
-
+		if resume > 0 && resp.StatusCode != 206 {
+			logger.Debugf("server does not support resume")
+			if _, err := w.Seek(0, os.SEEK_SET); err != nil {
```

Comment:
Should we also truncate the file to make sure that we really start from scratch?

Reply:
Interesting, I guess the concern is that the local file might be bigger than what we download and if we just seek here.  We only have a w io.ReadWriteSeeker here

I think this should be (and already is) caught in the higher level, i.e. don't try to resume if resume > downloadInfo.Size. But we need a test for this to be sure.

Reply:
Added the test now, the higher layers handle this correctly and will not do into download() if resume >= downloadInfo.Size.

Reply:
Another crazy idea so that we don't do unnecessary IO:
```
// hash was already fed what it needed

// read up to resume bytes
lr := &io.LimitedReader{R: resp.Body, N: resume}

// drop what we already have
io.Copy(ioutil.Discard, lr)
// TODO: verify that hash up to 'resume' bytes is identical

// ready to receive new data now as if  we got 206 Resume
```